pkg/utils/utils.go

```diff
@@ -656,3 +656,34 @@ func GetPhysPortName(name string) (string, error) {
 	}
 	return "", nil
 }
+
+func GetVFPciAddressFromIndex(pfPciAddress string, vfID int) (string, error) {
+	glog.V(2).Infof("GetVFPciAddressFromIndex(): PF %s, vfID %d", pfPciAddress, vfID)
+	vfDir := fmt.Sprintf("/sys/bus/pci/devices/%s/virtfn%d", pfPciAddress, vfID)
+	pciDevDir, err := os.Readlink(vfDir)
+	if err != nil {
+		return "", err
+	}
+
+	return path.Base(pciDevDir), nil
+}
+
+func GetVFNodeGUID(pciAddress string) (string, error) {
```

Comment:
need to see if we have netlink api for this

Reply:
It is possible to use rdmamap to get rdma device the use netlink to get node guid
```
rdmaDev, _ := netlink.RdmaLinkByName("mlx5_3")
fmt.Println(rdmaDev.Attrs.NodeGuid)
```

Reply:
Also from private function in rdmamap getNodeGuid